src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +80,10 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
I suggest using a named constant instead so that the code it's better documented. Seeing just 1000 here doesn't clarify what that means? max entities fetched?

I just checked the instructions and I see that they also put the value, but I recommend we improve on it. It's a valuable practice to learn.